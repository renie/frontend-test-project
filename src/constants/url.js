export const getAPIUrl = endpoint => `https://jsonplaceholder.typicode.com/${endpoint}`;
