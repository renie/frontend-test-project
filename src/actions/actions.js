import { getAPIUrl } from '../constants/url.js'
import {
  GET_USERS_REQUEST,
  GET_USERS_SUCCESS,
  GET_USERS_ERROR
} from '../constants/actionTypes.js'


const requestUsers = () => ({ type: GET_USERS_REQUEST });
const receiveUsers = users => ({ type: GET_USERS_SUCCESS, users });
const errorUsers = error => ({ type: GET_USERS_ERROR, error });

export const fetchUsers = () => dispatch => {
  dispatch(requestUsers());

  return fetch(getAPIUrl('users'))
      .then(
        response => response.json(),
        error => dispatch(errorUsers(error))
      )
      .then(
        json => dispatch(receiveUsers(json))
      )
      .catch(
        error => dispatch(errorUsers(error)))
}
