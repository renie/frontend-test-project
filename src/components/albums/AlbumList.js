import React, { Component } from 'react';
import { connect } from 'react-redux';
import Album from './Album'
import { fetchAlbums } from '../../actions/actions';
import './albumList.css'

export class AlbumList extends Component {
  componentDidMount() {
    this.props.fetchAlbums(this.props.userId)
  }

  render() {
    const user = this.props.users.find(user => user.id === this.props.userId)
    const albums = user ? user.albums : []

    return (
      <div className="albumListContainer">
        <h3 className="label">Albums:</h3>
        <ul className="albumList">
          {
            albums.map((album, index) => (
              <Album key={album.id} album={album} />
            ))
          }
        </ul>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  users: state.appReducer.app.users
})

const mapDispatchToProps = dispatch => ({
  fetchAlbums: userId => dispatch(fetchAlbums(userId))
})

export default connect(mapStateToProps, mapDispatchToProps)(AlbumList)
