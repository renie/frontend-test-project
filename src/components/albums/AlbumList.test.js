import React from 'react';
import ReactDOM from 'react-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';

import fetchMock from 'fetch-mock';
import { AlbumList } from './AlbumList'
import { fetchAlbums } from '../../actions/actions';
import { getAPIUrl } from '../../constants/url.js';


const mockStore = configureMockStore([thunk])

describe('AlbumList', () => {
  afterEach(() => {
    fetchMock.restore()
  });

  it('Create Component', () => {
    const expectedBody = [];
    const store = mockStore({ app: { users: [] }})
    const mockfetchAlbum = () => store.dispatch(fetchAlbums(id))
    const id = 1

    fetchMock.getOnce(getAPIUrl('albums', id), {
      body: expectedBody,
      headers: { 'content-type': 'application/json' }
    })

    const wrapper = shallow(<AlbumList store={store} fetchAlbums={mockfetchAlbum} userId={id} users={[]} />);

    expect(wrapper).toMatchSnapshot();
  });
});
