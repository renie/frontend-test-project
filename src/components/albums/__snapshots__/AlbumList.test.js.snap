// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumList Create Component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AlbumList
    fetchAlbums={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    userId={1}
    users={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="label"
        >
          Albums:
        </h3>,
        <ul
          className="albumList"
        />,
      ],
      "className": "albumListContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Albums:",
          "className": "label",
        },
        "ref": null,
        "rendered": "Albums:",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "albumList",
        },
        "ref": null,
        "rendered": Array [],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="label"
          >
            Albums:
          </h3>,
          <ul
            className="albumList"
          />,
        ],
        "className": "albumListContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Albums:",
            "className": "label",
          },
          "ref": null,
          "rendered": "Albums:",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "albumList",
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
